src/main/java/com/google/api/codegen/config/FixedResourceNameConfig.java

```diff
@@ -45,6 +45,25 @@ public ResourceNameType getResourceNameType() {
     return ResourceNameType.FIXED;
   }
 
+  @Nullable
+  public static FixedResourceNameConfig createFixedResourceNameConfig(
+      DiagCollector diagCollector, String entityName, String fixedValue, @Nullable ProtoFile file) {
+
+    if (entityName == null || fixedValue == null) {
```

Comment:
Rather than having this code be duplicated, can we have the original createFixedResourceNameConfig function call this one?

Reply:
even better